openwisp_users/admin.py

```diff
@@ -185,6 +185,16 @@ class Meta(BaseUserCreationForm.Meta):
     class Media:
         js = ('admin/js/jquery.init.js', 'openwisp-users/js/addform.js')
 
+    def clean_email(self):
```

Comment:
https://github.com/openwisp/openwisp-users/blob/114b16b51a415223cb2f34fd2a753bfe6a35d27f/openwisp_users/base/models.py#L50-L51

When the email is unique then it will automatically raise validation error @Saurav-Shrivastav.
I think it is not needed here.

Reply:
Django doesn't check for the case of the email, that is why I've added the validation in the form. (unique=True, is case-insensitive). If a new user is created from the admin-panel - it'll lead to `Integrity Error`, in order to handle that gracefully, the `clean_email` function is required.

Reply:
But in the `save` method you are making it lowercase. Then according to this, email with uppercase letters will always be converted into the lower case before saving, and when it will try to save pre-existing email it will raise validation error.
Correct me if I am wrong.

Reply:
When a request for adding a user is made, the validation takes place first and then the `save` method is reached at a later stage. So, the Validation at the form level will pass for the case insensitive email and it'll reach the `save` method, which will then throw an `IntegrityError: email already exists`.